lib/rubygems/config_file.rb

```diff
@@ -43,6 +44,7 @@ class Gem::ConfigFile
   DEFAULT_BULK_THRESHOLD = 1000
   DEFAULT_VERBOSITY = true
   DEFAULT_UPDATE_SOURCES = true
+  DEFAULT_CONCURRENT_DOWNLOADS = 8
```

Comment:
should this default to `nproc` instead of a fixed number?

Reply:
There's no reason to tie this to the number of processors/cores.

1. It's an I/O-bound task. I/O introduces a lot of waits, so even if you have a few more threads than the number of cores, it won't peg your CPU.
2. 16-core, 32-thread CPUs are a thing. `Etc.nprocessors` will return `32` on those systems. Using 32 threads for downloading files from a single server is a horrible, horrible idea.

There's also the fact that `Etc.nprocessors` doesn't work on Windows, so we'd have to roll our own thing for that.